grpc-protocol/src/main/java/com/linecorp/armeria/internal/common/grpc/protocol/GrpcStatusCodeUtil.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ */
+
+package com.linecorp.armeria.internal.common.grpc.protocol;
+
+public final class GrpcStatusCodeUtil {
+
+    /**
+     * gRPC uses the status codes from 0 to 16.
+     * <a href="https://grpc.github.io/grpc/core/md_doc_statuscodes.html">Status codes and their use in gRPC</a>
+     */
+    private static final String[] codeStrings = new String[17];
+
+    static {
+        for (int code = 0; code < codeStrings.length; code++) {
+            codeStrings[code] = String.valueOf(code);
+        }
+    }
+
+    public static String intToString(int code) {
+        return codeStrings[code];
```

Comment:
How about falling back to `Integer.toString()` if out of bounds?

Reply:
>  Can have an internal array with 256 or 64K entries reused everywhere needed.

I had a similar idea. Needs to generalize it.

Reply:
The pooled version looks 4~5 times faster.
```
IntegersBenchmark.pooledIntToString   thrpt    5  423959.144 ± 10357.257  ops/s
IntegersBenchmark.jdkIntToString      thrpt    5   95247.269 ±  1025.624  ops/s
```

I've also checked pooled version for string-to-int using `Object2IntOpenHashMap<String>`.
The lookup cost seems more expensive.
```
IntegersBenchmark.pooledStringToInt   thrpt    5   77978.511 ±  1866.730  ops/s
IntegersBenchmark.jdkStringToInt      thrpt    5  109831.946 ±  4592.673  ops/s
```

Reply:
Just to confirm, did you run on Java 9+? It seems Java 9 added a Hotspot intrinsic `_Integer_toString` so just wondering. Guessing it still allocates though so caching should be worth it.